pkg/fieldpath/paved.go

```diff
@@ -404,6 +400,12 @@ func (p *Paved) setValue(s Segments, value any) error {
 				return nil
 			}
 
+			if !final && s[i+1].Type == SegmentIndex &&
```

Comment:
Looks like it is already `not final` (see the previous if).

```suggestion
			if s[i+1].Type == SegmentIndex &&
```